src/Microsoft.Diagnostics.Monitoring.EventPipe/Traces/TraceEventExtensions.cs

```diff
@@ -0,0 +1,220 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Diagnostics.Tracing;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe
+{
+    internal static partial class TraceEventExtensions
+    {
+        [ThreadStatic]
+        private static KeyValuePair<string, object?>[]? s_TagStorage;
```

Comment:
Can this be string/string?

Reply:
It could be. Today everything pumped through `DiagnosticSourceEventSource` is in fact a string. But I went this direction because that is sort of an issue. When it comes to OTel distributed tracing, there are semantic conventions. Something like `http.response.status_code` _should_ be an int to be compliant with the spec. I went with `object` so it could be fixed in the future if we add some typing info to the events or perhaps some conversion in the pipeline 🤷